### PR TITLE
fix(server): honor sampling parameters exactly

### PR DESCRIPTION
**Pull request by bstnxbt**
## Summary
- keep greedy requests on DFlash
- route stochastic sampling, logits processing, and logprobs through exact target-only MLX-LM generation
- preserve all supported generation fields through `/v1/responses`

## Validation
- `PYTHONPATH=$PWD pytest tests/test_serve_state_machine.py tests/test_responses_adapter.py -q` — 55 passed
- `PYTHONPATH=$PWD pytest tests/ -q` — 1252 passed, 11 skipped
- xhigh review: COMMIT-READY yes

### CODE REVIEW

**Comment by bstnxbt** [comment]
Fixed in [`058c886`](https://github.com/bstnxbt/dflash-mlx/commit/058c886): greedy DFlash is unchanged, while temperature/top-k/top-p/min-p, penalties, XTC, logit bias, and logprobs now use exact target-only MLX-LM semantics. Verified by 55 focused tests and the full 1252-test suite.
